src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -66,16 +66,27 @@ uiModules
     location: 'Visualization Editor'
   });
 
+  // Retrieve the resolved SavedVis instance.
   const savedVis = $route.current.locals.savedVis;
 
+  // Instance of vis.js.
   const vis = savedVis.vis;
+
+  // Clone the _vis instance.
   const editableVis = vis.createEditableVis();
+
+  // NOTE: Why is this instance method being overwritten?
   vis.requesting = function () {
     const requesting = editableVis.requesting;
+    // Invoking requesting() calls onRequest on each agg's type param.
+    // NOTE: What is the structure of an agg_config, and the role of type
```

Comment:
The `AggConfig` objects represent the aggs defined in the left-hand nav of the visualize app. Some of those `AggConfig` objects need to know properties about the context of the app at the time that the visualization is requested, like the [date histogram for instance](http://github.com/spalger/kibana/blob/01cbb4765a3bab539b2d3abeb60b0ac8af304618/src/ui/public/agg_types/buckets/date_histogram.js#L95) which tracks the bounds of the time filter at the time of each request.

When a vis is marked as being requested, the bounds of that vis are updated and the corresponding data can be resolved based on new bounds.